qa/integration-tests/src/test/java/io/zeebe/broker/it/clustering/ClusteredDataDeletionTest.java

```diff
@@ -90,22 +94,19 @@ public void shouldDeleteDataOnLeader() {
     final int leaderNodeId = clusteringRule.getLeaderForPartition(1).getNodeId();
     final Broker leader = clusteringRule.getBroker(leaderNodeId);
 
-    while (getSegmentsCount(leader) <= 2) {
-      clusteringRule
-          .getClient()
-          .newPublishMessageCommand()
-          .messageName("msg")
-          .correlationKey("key")
-          .send()
-          .join();
-    }
+    fillSegments(List.of(leader), SEGMENT_COUNT);
 
     // when
-    final var segmentCount =
+    final var segmentCountBeforeSnapshot =
         takeSnapshotAndWaitForReplication(Collections.singletonList(leader), clusteringRule);
 
     // then
-    TestUtil.waitUntil(() -> getSegments(leader).size() < segmentCount.get(leaderNodeId));
+    waitUntil(
```

Comment:
Nit: 
```suggestion
  Awaitility.await()
        .untilAsserted(
            () ->
                assertThat(getSegmentsCount(leader))
                    .isLessThan(segmentCountBeforeSnapshot.get(leaderNodeId)));
```

Reply:
Do you suggest to replace our `TestUtil`? :sweat_smile:

Reply:
:D We had some discussions before about starting to use Awaitility instead of our own `waitUntil`. This looks like a good place to use it.

Reply:
We already use it for example here https://github.com/zeebe-io/zeebe/blob/379afe7174267f168d967b09d295e5b5ab0fd6c4/atomix/cluster/src/test/java/io/atomix/cluster/protocol/SwimProtocolTest.java#L385-L387

Reply:
Ok :+1: If we plan to replace the `TestUtil` then we should create an issue and communicate it to the team.